services/hierarchy/src/main/kotlin/RepositoryService.kt

```diff
@@ -119,4 +121,60 @@ class RepositoryService(
     ): Repository = db.dbQuery {
         repositoryRepository.update(repositoryId, type, url)
     }
+
+    /**
+     * Remove a user from a group that grant roles and permissions on repository level.
+     */
+    suspend fun removeUserFromGroup(
+        username: String,
+        repositoryId: Long,
+        groupId: String
+    ) {
+        getRepository(repositoryId)
+            ?: throw ResourceNotFoundException("Repository with repositoryId '$repositoryId' not found.")
+
+        val repositoryRole = try {
+            RepositoryRole.valueOf(groupId.uppercase().removeSuffix("S"))
+        } catch (e: IllegalArgumentException) {
+            throw ResourceNotFoundException(
+                "Group with groupId '$groupId' not found. Must be one of ${RepositoryRole.entries.toJoinedString()}",
+                e
+            )
+        }
+
+        val groupName = repositoryRole.groupName(repositoryId)
+
+        // As the AuthorizationService does not distinguish between technical exceptions (e.g. cannot connect to
+        // Keycloak) and business exceptions (e.g. user not found), we can't do special exception handling here
+        // and just let the exception propagate.
+        authorizationService.removeUserFromGroup(username, groupName)
+    }
+
+    /**
+     * Add a user to one of the three groups that grant roles and permissions on repository level.
+     */
+    suspend fun addUserToGroup(
+        username: String,
+        repositoryId: Long,
+        groupId: String
+    ) {
+        getRepository(repositoryId)
+            ?: throw ResourceNotFoundException("Repository with repositoryId '$repositoryId' not found.")
+
+        val repositoryRole = try {
+            RepositoryRole.valueOf(groupId.uppercase().removeSuffix("S"))
+        } catch (e: IllegalArgumentException) {
+            throw ResourceNotFoundException(
+                "Group with groupId '$groupId' not found. Must be one of ${RepositoryRole.entries.toJoinedString()}",
+                e
+            )
+        }
+
+        val groupName = repositoryRole.groupName(repositoryId)
+
+        // As the AuthorizationService does not distinguish between technical exceptions (e.g. cannot connect to
+        // Keycloak) and business exceptions (e.g. user not found), we can't do special exception handling here
+        // and just let the exception propagate.
+        authorizationService.addUserToGroup(username, groupName)
+    }
```

Comment:
There should be tests for the new functions, similar to those in `OrganizationServiceTest`.